### PR TITLE
Extension refactored to adapt to GNOME 45

### PR DESCRIPTION
**Pull request by pratap-panabaka**
Hi,

I tried to adapt the extension from GNOME 44 code to GNOME 45.
I could not make the location in the Prefs window to choose from drop-down so used the spin button same as index selection.

image

Thanks

### CODE REVIEW

**Review by jancborchardt — COMMENTED** [review]
Looks nice, can’t test on GNOME 45 though. :) 

@PRATAP-KUMAR what do you think about using a settings cog icon on the right of the "Write One Thing …" input field for getting to settings instead of the dedicated line below? That would look a bit simpler, but I’m not sure if it is ok according to the GNOME HIG.

**Comment by kra-mo** [comment]
"You can always access it in Extension Manager" should probably be a description, not part of the title. Extension Manager also isn't an official GNOME app, so it should say Extensions.

**Comment by kra-mo** [comment]
@jancborchardt The HIG doesn't extend to the Shell, but having an icon next to text is common practice around other UI elements there, so I think it's fine in that regard.

**Comment by kra-mo** [comment]
@jancborchardt The HIG doesn't extend to the Shell, but I don't see why it would go against it. Having a simple icon for common actions like this is preferable to text, so I think it's fine in that regard.

**Comment by pratap-panabaka** [comment]
Hi, I tried to incorporate above discussion points.

image

**Comment by kra-mo** [comment]
I've pushed some UI changes to the branch. This is what it looks like now:

Screenshot from 2023-10-20 15-54-38

Screenshot from 2023-10-20 15-54-42

Otherwise LGTM

**Review by kra-mo — APPROVED** [review]

**Comment by pratap-panabaka** [comment]
Looks Nice. If possible plz make changes to Location Prefs to Left, Center, Right. I am not able to achieve that.

Thanks

**Comment by kra-mo** [comment]
@PRATAP-KUMAR Done.

image


Though I am not very proficient with JavaScript, so if you think you could clean the code up, go ahead.

**Comment by pratap-panabaka** [comment]
Hi,
I have updated Prefs to have option for Emojis via Adw.EntryRow.
Also added `Prefs` via Context menu along with copy, paste.

Any changes you feel, feel free to change and look to upload to gnome-extensions.

Thanks

image

**Comment by kra-mo** [comment]
Having `<primary>,` open an emoji picker is standard across GNOME, so there really doesn't need to be text explaining it. I pushed a commit that removes it and updates the screenshot.

Screenshot from 2023-10-21 13-07-50

**Comment by pratap-panabaka** [comment]
That is nice. I hope the owner of extension can now try to submit it for review from gnome-extensions.

File: schemas/gschemas.compiled
Comment (dantehemerson) [inline schemas/gschemas.compiled:1]: This binary file don't needs to be added to git, it's only added to the published extension.

File: metadata.json
Comment (dantehemerson) [inline metadata.json]: Does this line need to be included?

File: .gitignore
Comment (dantehemerson) [inline .gitignore]: I would like to leave these folders and files ommited, only to not deploy by mistake after we generate the extension to publish.

**Review by dantehemerson — COMMENTED** [review]
@PRATAP-KUMAR @kra-mo Thanks for these changes!
I just leave a couple of comments:

**Review by dantehemerson — APPROVED** [review]
LGTM!

File: schemas/gschemas.compiled
Comment (pratap-panabaka) [reply schemas/gschemas.compiled:1]: Hey it is not necessary to add even while submitting to the gnome-extensions for review, since gnome-shell v44 it will be generated at the time of installation.

File: metadata.json
Comment (pratap-panabaka) [reply metadata.json]: Hey it is not necessary, how ever when we submit to extensions for review, that line is auto generated.

File: .gitignore
Comment (pratap-panabaka) [reply .gitignore]: @dantehemerson I have removed `_generated` line from metadata and added those two lnes to `git ignore`.  You proceed to upload to gnome-extensions. Thanks

**Comment by kra-mo** [comment]
@jancborchardt your verdict?

**Comment by dantehemerson** [comment]
I sent the new version for review: https://extensions.gnome.org/review/47068

**Comment by pratap-panabaka** [comment]
> I sent the new version for review: https://extensions.gnome.org/review/47068

Nice. Lets wait for the approval. Hope there wont be any comments from the reviewer. If anything please share.

**Comment by pratap-panabaka** [comment]
Hi, I have renamed the branch master to main.

image

```
git branch -m master main
git fetch origin
git branch -u origin/main main
git remote set-head origin -a
```

**Comment by dantehemerson** [comment]
@PRATAP-KUMAR @kra-mo We received a comment in the review: https://extensions.gnome.org/review/47068

**Comment by pratap-panabaka** [comment]
Ok, but the reviewer is saying about right panel, what if the user chooses to be at left or at the middle? do you have any idea @dantehemerson

**Comment by pratap-panabaka** [comment]
@dantehemerson I have made the changes and pushed to main branch. Tested the extension. please have a look. Thanks

image

**Comment by dantehemerson** [comment]
@PRATAP-KUMAR I don't have gnome 45, I cannot test. But looks good to me. I will send for review.

**Comment by dantehemerson** [comment]
@PRATAP-KUMAR I just have this question: https://github.com/one-thing-gnome/one-thing/commit/c4f415f5c6ac03f89dd7ec2c51479674178f86c3#r131123642

**Comment by pratap-panabaka** [comment]
@dantehemerson I have just tested with else statement and the extension is not working as expected. I understood what you mean. We may have to dig more if we want to go with else clause, I think `actorAddedSignal` is causing the issue. Shall we dig more on this or submit for review with the latest commit I made?

**Pull request merged**
